Sources/Rendering/Core/ColorTransferFunction/index.js

```diff
@@ -871,6 +871,21 @@ function vtkColorTransferFunction(publicAPI, model) {
     return model.table;
   };
 
+  publicAPI.buildFunctionFromTableLong = (size, table) => {
+    publicAPI.removeAllPoints();
+    for (let i = 0; i < size; i++) {
+      const node = {
+        x: table[i * 6],
+        r: table[i * 6 + 1],
+        g: table[i * 6 + 2],
+        b: table[i * 6 + 3],
+        midpoint: table[i * 6 + 4],
+        sharpness: table[i * 6 + 5],
+      };
+      model.nodes.push(node);
+    }
+    publicAPI.sortAndUpdateRange();
+  };
```

Comment:
If `table` was a `vtkDataArray`, we could get the number of components per tuple:
- 3 ->RGB
- 4 ->XRGB
- 5 -> RGBMS
- 6 -> XRGBMS

You would then have something very generic.
(here is how you could convert `table` into a `vtkDataArray`:
```
ctf.buildFunctionFromArray(vtkDataArray.newInstance({values: table, numberOfComponents: 6})
```

Reply:
makes sense, I will make that change